core/DataAccess/TableMetadata.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\DataAccess;
+
+use Piwik\Db;
+
+/**
+ * Data Access Object that can be used to get metadata information about
+ * the MySQL tables Piwik uses.
+ */
+class TableMetadata
```

Comment:
I'm probably picking but I believe this kind of code would be better in `Piwik\Db`, i.e.

- `Piwik\Db` for code related to database, unrelated to domain stuff
- `Piwik\DataAccess` for code related to linking domain stuff to the db

To me getting the columns of a table is really DBAL stuff. This class could still be separated from the `Db` class (I see you deprecated the method), maybe in a `Piwik\Db\Schema` class (like the [`Schema`](https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Schema/Schema.php) class in Doctrine)? I imagine later we could have a full abstraction of schema manipulation in there (again, like in Doctrine) to get/create tables, columns, etc…

Thoughts?